projects/components/src/sequence/renderer/sequence-bar-renderer.service.ts

```diff
@@ -175,6 +207,49 @@ export class SequenceBarRendererService {
     options.selected = options.selected === dataRow ? undefined : dataRow;
     options.onSegmentSelected(dataRow);
   }
+
+  private setupMarkerHoverListener(plotSelection: SequenceSVGSelection, options: SequenceOptions): void {
+    plotSelection
+      .selectAll<SVGGElement, Marker>(`g.${SequenceBarRendererService.MARKER_CLASS}`)
+      .on('mouseenter', (dataRow, index, nodes) => {
+        options.onMarkerHovered({ marker: dataRow, origin: nodes[index].getBoundingClientRect() });
+      });
+  }
+
+  private getGroupedMarkers(segment: SequenceSegment, xScale: ScaleLinear<number, number>): Marker[] {
+    const scaledStart: number = Math.floor(xScale(segment.start)!);
+    const scaledEnd: number = Math.floor(xScale(segment.end)!);
+    const pixelScaledMarkers: Marker[] = segment.markers.map((marker: Marker) => ({
+      ...marker,
+      markerTime: Math.floor(xScale(marker.markerTime)!)
+    }));
+    const scaledNormalizedMarkers: Marker[] = [];
+    let markerTime = -1 * Infinity;
+    let index = -1;
+    pixelScaledMarkers.forEach((marker: Marker) => {
```

Comment:
nit: With a later PR, if we can make this grouping logic a little easier to understand then it would be great. You can break it down into multiple smaller 'english like' functions :)